ee/agent/knapsack/knapsack.go

```diff
@@ -59,6 +62,18 @@ func New(stores map[storage.Store]types.KVStore, flags types.Flags, db *bbolt.DB
 	return k
 }
 
+// SetRunID sets the run ID in the knapsack
+func (k *knapsack) SetRunID(id string) {
+	runID = id
+	k.slogger.With("run_id", id)
+	k.systemSlogger.With("run_id", id)
```

Comment:
Do we need e.g. --

```golang
k.slogger = k.slogger.With("run_id", id)
k.systemSlogger = k.systemSlogger.With("run_id", id)
```

? I think `.With` creates and then modifies a clone, so it wouldn't update k.slogger.

Reply:
Yeah, I did that in launcher but missed in Knapsack. I have pushed out another commit to fix this with the assignment.